pkg/controller/installation/installation_controller.go

```diff
@@ -377,12 +391,265 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		return retryRequeue, nil
 	}
 	installation.Status.Stage = integreatlyv1alpha1.StageName("complete")
+
+	if (isFirstInstallReconcile(installation) || isUpgradeReconcile(installation)) && allProductsReconciled(installation) {
+		installation.Status.Version = installation.Status.ToVersion
+		installation.Status.ToVersion = ""
+		metrics.SetRhmiVersions(string(installation.Status.Stage), installation.Status.Version, installation.Status.ToVersion)
+	}
+
 	_ = r.client.Status().Update(r.context, installation)
 	metrics.RHMIStatusAvailable.Set(1)
 	logrus.Infof("installation completed succesfully")
 	return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Minute}, nil
 }
 
+func isFirstInstallReconcile(installation *integreatlyv1alpha1.RHMI) bool {
+	return installation.Status.Version == ""
+}
+
+func isUpgradeReconcile(installation *integreatlyv1alpha1.RHMI) bool {
+	status := installation.Status
+	if status.ToVersion != "" {
+		return true
+	}
+	return false
+}
+
+func allProductsReconciled(installation *integreatlyv1alpha1.RHMI) bool {
+	stages := installation.Status.Stages
+
+	return (verifyClusterRHSSOVersions(stages) &&
+		verifyCloudResourceVersions(stages) &&
+		verifyAMQOnlineVersions(stages) &&
+		verifyApicuritoVersions(stages) &&
+		verifyCodeReadyWorkspacesVersions(stages) &&
+		verifyFuseOnOpenshiftVersions(stages) &&
+		verifyMonitoringVersions(stages) &&
+		verify3ScaleVersions(stages) &&
+		verifyFuseOnlineVersions(stages) &&
+		verifyDataSyncVersions(stages) &&
+		verifyRHSSOUserVersions(stages) &&
+		verifyUPSVersions(stages) &&
+		verifySolutionExplorerVersions(stages))
+}
+
+func verifySolutionExplorerVersions(stages map[integreatlyv1alpha1.StageName]integreatlyv1alpha1.RHMIStageStatus) bool {
```

Comment:
This comment applies to all the product verify functions. If the product uses floating tags and there is a patch release, does this function still work as expected? I feel this will still compare the old versions and not see the newer version.